regen/regen_lib.pl

```diff
@@ -130,7 +130,14 @@ sub read_only_top {
         unless exists $lang_opener{$lang};
     my $style = $args{style} ? " $args{style} " : '   ';
 
-    my $raw = "-*- buffer-read-only: t -*-\n";
+    # Generate the "modeline" for syntax highlighting based on the language
+    my $raw;
+    if ($lang eq 'C') {
+        $raw = "-*- mode: C; buffer-read-only: t -*-\n"; # see line 40 where $lang is set: affects .c, .h, .inc, .tab, and .act files
+    }
+    else {
+        $raw = "-*- mode: Perl; buffer-read-only: t -*-\n"; # ibid. ".pod" being the only other one which is in Perl mode as well
+    }
```

Comment:
Given that `$lang` can only be one of `Perl`, `Pod`, or `C`, and Emacs has `pod-mode`, why not unconditionally add `mode: $lang;`?

Reply:
Yes, I thought about that as well.
I'm not sure `Pod` will be accepted by GitHub.
I'm not aware if Emacs had pod-mode.

Reading [perl-mode](https://github.com/emacs-mirror/emacs/blob/master/lisp/progmodes/perl-mode.el) and [cperl-mode](https://github.com/emacs-mirror/emacs/blob/master/lisp/progmodes/cperl-mode.el) I don't find that easily.

Maybe you mean https://www.emacswiki.org/emacs/PodMode (which might not be in Emacs core being available as an extension)?

If that's really the case I we can do that, yes.

Reply:
GitHub does recognise `-*- mode: pod -*-` (I just tested by replacing the vim modeline in `INSTALL` on a branch).

You're right, `pod-mode` is not included with Emacs itself, but I expect anyone who's hacking Perl using Emacs and cares about syntax highlighting to have it installed to get highlighting in `.pod` files. I already had it because Debian's `emacs-goodies-el` pulls it in.

Reply:
Good, I will modify the code to be shorter then.